packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -460,6 +468,18 @@ class NavigatorState extends State<Navigator> {
     return _history.length > 1 || _history[0].willHandlePopInternally;
   }
 
+  void didStartUserGesture() { config.observer?.didStartUserGesture(); }
```

Comment:
By convention, foo.didDoBar() methods are called by the owner of the foo object to notify it that it did bar. (So on the observer class it makes sense.)
This API is more about notifying a peer object (the navigator) that something happened, and asking it to do something about it. So I'd go with something more like "startGesture".